### PR TITLE
Ignore similar chars in random_string

### PR DESCRIPTION
**Pull request by gbyx3**
##### SUMMARY
Added the option to ignore certain characters.
With the current parameters it is not possible to avoid similar character while also improve readability for passwords. If the parameter “override_all” is used the parameters “min_lower”, “min_upper”, “min_numeric” and “min_special” are ignored. This makes it hard to both achieve readability for passwords and adhere to configured password policies.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/lookup/random_string.py

##### ADDITIONAL INFORMATION

```
lookup('random_string', ignore_similar_chars=True, similar_chars='iIlLoO01')
```

### CODE REVIEW

File: plugins/lookup/random_string.py [hunk 1]
```diff
@@ -80,6 +80,16 @@
         - Override all values of O(numbers), O(upper), O(lower), and O(special) with
           the given list of characters.
         type: str
+      ignore_similar_chars:
+        description:
+        - Ignore similar characters, such as l and 1, or O and 0.
```
Comment (felixfontein) [inline plugins/lookup/random_string.py:85]:
```suggestion
        - Ignore similar characters, such as V(l) and V(1), or V(O) and V(0).
        - These characters can be configured in O(similar_chars).
```

File: plugins/lookup/random_string.py [hunk 2]
```diff
@@ -80,6 +80,16 @@
...
+        default: false
+        type: bool
```
Comment (felixfontein) [inline plugins/lookup/random_string.py:87]:
```suggestion
        type: bool
        version_added: 7.5.0
```

File: plugins/lookup/random_string.py [hunk 3]
```diff
@@ -80,6 +80,16 @@
...
+      similar_chars:
+        description:
+        - Overide a list of characters not to be use in the string.
+        default: "il1LoO0"
+        type: str
```
Comment (felixfontein) [inline plugins/lookup/random_string.py:92]:
```suggestion
        type: str
        version_added: 7.5.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments.

**Comment by gbyx3** [comment]
Thanks! Let me know if you want me to commit these changes.

**Comment by felixfontein** [comment]
@gbyx3 please do so (also feel free to adjust the description for `ignore_similar_chars`).

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this somewhen next week.

**Comment by felixfontein** [comment]
@gbyx3 thanks for your contribution!

**Pull request merged**